lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3SseRequestConfigurator.java

```diff
@@ -39,6 +49,47 @@ public static void addEncryptionSettings(CreateMultipartUploadRequest.Builder bu
             case NONE -> { /* ignored */ }
             case S3 -> builder.serverSideEncryption(AES256);
             case KMS -> builder.serverSideEncryption(AWS_KMS).ssekmsKeyId(context.sseKmsKeyId());
+            case CUSTOMER -> {
+                builder.sseCustomerAlgorithm(context.sseCustomerKey().algorithm());
+                builder.sseCustomerKey(context.sseCustomerKey().key());
+                builder.sseCustomerKeyMD5(context.sseCustomerKey().md5());
+            }
+        }
+    }
+
+    public static void addEncryptionSettings(CompleteMultipartUploadRequest.Builder builder, S3SseContext context)
+    {
+        if (context.sseType() == CUSTOMER) {
+            builder.sseCustomerAlgorithm(context.sseCustomerKey().algorithm());
+            builder.sseCustomerKey(context.sseCustomerKey().key());
+            builder.sseCustomerKeyMD5(context.sseCustomerKey().md5());
```

Comment:
Let's extract `context.sseCustomerKey()` and use it the chain.
```suggestion
            S3SseCustomerKey  customerKey  = context.sseCustomerKey();
            builder.sseCustomerAlgorithm(customerKey.algorithm()).sseCustomerKey(customerKey.key()).sseCustomerKeyMD5(customerKey.md5());
```